base_optional_create_edit/models/ir_ui_view.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+
+from odoo import api, models
+
+
+class Model(models.AbstractModel):
+    _inherit = "base"
+
+    @api.model
+    def get_views(self, views, options=None):
+        res = super().get_views(views, options)
+        views = res.get("views")
+        view_form = views["form"]["arch"]
+        tree = etree.fromstring(view_form)
+        view_fields = set(tree.xpath(".//field[not(ancestor::field)]"))
+
+        field_names = []
+        for field_name, model in self._fields.items():
```

Comment:
For each relational field on the view, you'll execute a `search(...)`.
Please find another solution to do it only once :pray: 
And you can improve your domain to include your new field `avoid_create_edit`.

IMO your domain should be something like (it's just a draft):
```
comodel_names = [model.comodel_name for field_name, model in self._fields.items() if model.relational]
domain = [('avoid_create_edit', '=', True), ('model', 'in', comodel_names)]
models = self.env["ir.model"].search(domain)
fields_names = [field_name for field_name, model in self._fields.items() if model.comodel_name in models.model]
```